### PR TITLE
NMS-12081: System tests for thresholding

### PR DESCRIPTION
**Pull request by j-white**
JIRA: https://issues.opennms.org/browse/NMS-12081

Here we add a new system test used to verify thresholding of response time (pollerd) and generic resources (collectd w/ JDBC collector).

We also update the system tests APIs to be able to control the file permissions for overlaid files, so that we can set the executable bit on our script.

### CODE REVIEW

File: smoke-test/src/main/java/org/opennms/smoketest/utils/RestClient.java [hunk 1]
```diff
@@ -104,6 +109,16 @@ public RestClient(URL url, String username, String password) {
...
     }
 
+    /**
+     * Create a REST client that connects to http://localhost:8980/ with the default credentials.
+     * Used for testing.
+     *
+     * @return a new REST client for OpenNMS on the localhost
+     */
+    public static RestClient forLocalhost() {
+        return new RestClient(InetSocketAddress.createUnresolved("127.0.0.1", 8980));
```
Comment (mvrueden) [inline smoke-test/src/main/java/org/opennms/smoketest/utils/RestClient.java]: 🤸‍♀

File: smoke-test/src/test/java/org/opennms/smoketest/ThresholdingIT.java [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    private static final String TEST_NODE_FS = "test";
+    private static final String TEST_NODE_FID = "thresholding-test-node";
+    private static final String TEST_NODE_CRITERIA = TEST_NODE_FS + ":" + TEST_NODE_FID;
+    private static final String TEST_NODE_CATEGORY = "ThresholdingTest";
+    private static final String TEST_SVC_NAME = "SvcToThreshold";
+
+    @ClassRule
+    public static final OpenNMSStack stack = OpenNMSStack.withModel(StackModel.newBuilder()
+            .withOpenNMS(OpenNMSProfile.newBuilder()
+                    .withFile(getResource("thresholding/poller-configuration.xml"), "etc/poller-configuration.xml")
```
Comment (mvrueden) [inline smoke-test/src/test/java/org/opennms/smoketest/ThresholdingIT.java]: Is it worth trying to do the `getResouce(...)` call in `withFile(String, String)` instead as it is always performed anyways?

**Comment by mvrueden** [comment]
LGTM, one minor suggestion. I like the OverlayFile and OverlayUtils 💯

File: smoke-test/src/test/java/org/opennms/smoketest/ThresholdingIT.java (hunk 2 above)
Comment (j-white) [reply smoke-test/src/test/java/org/opennms/smoketest/ThresholdingIT.java]: Good point, this will be pretty common, so worth adding there.

**Pull request closed without merging**